python/amici/sbml_import.py

```diff
@@ -1499,6 +1501,50 @@ def _replace_special_constants(self) -> None:
                     f'Encountered currently unsupported element id {constant}!'
                 )
 
+    def _replace_reactions_in_rule_formula(self, rule, formula):
+        """
+        SBML allows reaction IDs in rules, which should be interpreted as the
+        reaction rate. However, an assignment or rate "...rule cannot be
+        defined for a species that is created or destroyed in a reaction,
+        unless that species is defined as a boundary condition in the model."
+
+        Here, valid SBML is assumed, so the above restriction is not checked.
+
+        :param rule:
+            The rule, (e.g. an element of the object returned by
+            `self.sbml.getListOfRules()`).
+
+        :param formula:
+            A rule formula that has already been parsed.
+            TODO create a function to parse rule formulae, as this logic is
+                 repeated a few times.
+
+        :return:
+            The substituted formula.
+        """
+        reaction_ids = [r.getId()
+                        for r in list(self.sbml.getListOfReactions())]
```

Comment:
```suggestion
                        for r in self.sbml.getListOfReactions()]
```
does self.sbml.getListOfReactions() not yield an `Iterable`?